Junit Jupiter Nested classes

When using `@WireMockTest` with two or more nested classes WireMock server never become running after first nested test class is executed.

WireMock: `wiremock-jre8: 2.31.0`

Here is demo:

```java
import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
import com.github.tomakehurst.wiremock.junit5.WireMockTest;
import org.junit.jupiter.api.Nested;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@WireMockTest
class WiremockNested {
    @Nested
    class A {
        @Test
        void a(WireMockRuntimeInfo info) {
            assertDoesNotThrow(info::getHttpBaseUrl);
        }
    }

    @Nested
    class B {
        @Test
        void b(WireMockRuntimeInfo info) {
            assertDoesNotThrow(info::getHttpBaseUrl);
        }
    }
}
```

Resulting in exception:
```
org.opentest4j.AssertionFailedError: Unexpected exception thrown: java.lang.IllegalStateException: Not listening on HTTP port. Either HTTP is not enabled or the WireMock server is stopped.

	at org.junit.jupiter.api.AssertDoesNotThrow.createAssertionFailedError(AssertDoesNotThrow.java:83)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:76)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:59)
	at org.junit.jupiter.api.Assertions.assertDoesNotThrow(Assertions.java:3195)
	at com.demo.WiremockNested$A.a(WiremockNested.java:16)
	<ommited>
Caused by: java.lang.IllegalStateException: Not listening on HTTP port. Either HTTP is not enabled or the WireMock server is stopped.
	at com.google.common.base.Preconditions.checkState(Preconditions.java:510)
	at com.github.tomakehurst.wiremock.WireMockServer.port(WireMockServer.java:187)
	at com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo.getHttpPort(WireMockRuntimeInfo.java:32)
	at com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo.getHttpBaseUrl(WireMockRuntimeInfo.java:48)
	at org.junit.jupiter.api.AssertDoesNotThrow.assertDoesNotThrow(AssertDoesNotThrow.java:72)
	... 80 more
```

--- a/src/main/java/com/github/tomakehurst/wiremock/junit5/WireMockExtension.java
+++ b/src/main/java/com/github/tomakehurst/wiremock/junit5/WireMockExtension.java
@@ -26,8 +26,6 @@ import org.junit.platform.commons.support.AnnotationSupport;
 
 import java.util.Optional;
 
-import static com.google.common.base.MoreObjects.firstNonNull;
-
 public class WireMockExtension extends DslWrapper implements ParameterResolver, BeforeEachCallback, BeforeAllCallback, AfterEachCallback, AfterAllCallback {
 
     private static final Options DEFAULT_OPTIONS = WireMockConfiguration.options().dynamicPort();
@@ -79,33 +77,48 @@ public class WireMockExtension extends DslWrapper implements ParameterResolver, 
 
     private void startServerIfRequired(ExtensionContext extensionContext) {
         if (wireMockServer == null) {
-            wireMockServer = new WireMockServer(resolveOptions(extensionContext));
-            wireMockServer.start();
+            startNewServer(resolveOptions(extensionContext));
+        } else if (!wireMockServer.isRunning()) {
+            Optional<WireMockTest> annotation = findWireMockTestAnnotation(extensionContext);
 
-            this.admin = wireMockServer;
-            this.stubbing = wireMockServer;
-
-            if (configureStaticDsl) {
-                WireMock.configureFor(new WireMock(this));
+            if (annotation.isPresent()) {
+                startNewServer(buildOptionsFromWireMockTestAnnotation(annotation.get()));
+            } else {
+                wireMockServer.start();
             }
+        }
+    }
+
+    private void startNewServer(Options options) {
+        wireMockServer = new WireMockServer(options);
+        wireMockServer.start();
+
+        this.admin = wireMockServer;
+        this.stubbing = wireMockServer;
+
+        if (configureStaticDsl) {
+            WireMock.configureFor(new WireMock(this));
         }
     }
 
     private void setAdditionalOptions(ExtensionContext extensionContext) {
         if (proxyMode == null) {
-            proxyMode = extensionContext.getElement()
-                    .flatMap(annotatedElement -> AnnotationSupport.findAnnotation(annotatedElement, WireMockTest.class))
+            proxyMode = findWireMockTestAnnotation(extensionContext)
                     .<Boolean>map(WireMockTest::proxyMode)
                     .orElse(false);
         }
     }
 
     private Options resolveOptions(ExtensionContext extensionContext) {
-        return extensionContext.getElement()
-                .flatMap(annotatedElement -> AnnotationSupport.findAnnotation(annotatedElement, WireMockTest.class))
+        return findWireMockTestAnnotation(extensionContext)
                 .<Options>map(this::buildOptionsFromWireMockTestAnnotation)
                 .orElse(Optional.ofNullable(this.options)
                                 .orElse(DEFAULT_OPTIONS));
+    }
+
+    private Optional<WireMockTest> findWireMockTestAnnotation(ExtensionContext extensionContext) {
+        return extensionContext.getElement()
+                .flatMap(annotatedElement -> AnnotationSupport.findAnnotation(annotatedElement, WireMockTest.class));
     }
 
     private Options buildOptionsFromWireMockTestAnnotation(WireMockTest annotation) {

--- a/src/test/java/com/github/tomakehurst/wiremock/junit5/JunitJupiterExtensionDeclarativeWithConfiguredNestedTest.java
+++ b/src/test/java/com/github/tomakehurst/wiremock/junit5/JunitJupiterExtensionDeclarativeWithConfiguredNestedTest.java
@@ -1,0 +1,56 @@
+/*
+ * Copyright (C) 2011 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.junit5;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+@WireMockTest(httpPort = 8765)
+class JunitJupiterExtensionDeclarativeWithConfiguredNestedTest {
+    @Test
+    void runs_on_the_supplied_port(WireMockRuntimeInfo wmRuntimeInfo) {
+        assertThat(wmRuntimeInfo.getHttpPort(), is(8765));
+    }
+
+    @Nested
+    @WireMockTest(httpPort = 8766)
+    class RunsOn8766 {
+        @Test
+        void runs_on_the_supplied_port(WireMockRuntimeInfo wmRuntimeInfo) {
+            assertThat(wmRuntimeInfo.getHttpPort(), is(8766));
+        }
+    }
+
+    @Nested
+    @WireMockTest(httpPort = 8767)
+    class RunsOn8767 {
+        @Test
+        void runs_on_the_supplied_port(WireMockRuntimeInfo wmRuntimeInfo) {
+            assertThat(wmRuntimeInfo.getHttpPort(), is(8767));
+        }
+    }
+
+    @Nested
+    class RunsOnInheritedPort {
+        @Test
+        void runs_on_the_supplied_port(WireMockRuntimeInfo wmRuntimeInfo) {
+            assertThat(wmRuntimeInfo.getHttpPort(), is(8765));
+        }
+    }
+}

--- a/src/test/java/com/github/tomakehurst/wiremock/junit5/JunitJupiterExtensionDeclarativeWithNestedTest.java
+++ b/src/test/java/com/github/tomakehurst/wiremock/junit5/JunitJupiterExtensionDeclarativeWithNestedTest.java
@@ -1,0 +1,54 @@
+/*
+ * Copyright (C) 2011 Thomas Akehurst
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.tomakehurst.wiremock.junit5;
+
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+@WireMockTest
+class JunitJupiterExtensionDeclarativeWithNestedTest {
+    @Test
+    void runs(WireMockRuntimeInfo info) {
+        assertDoesNotThrow(info::getHttpBaseUrl);
+    }
+
+    @Nested
+    class First {
+        @Test
+        void runs(WireMockRuntimeInfo info) {
+            assertDoesNotThrow(info::getHttpBaseUrl);
+        }
+    }
+
+    @Nested
+    class Second {
+        @Test
+        void runs(WireMockRuntimeInfo info) {
+            assertDoesNotThrow(info::getHttpBaseUrl);
+        }
+    }
+
+    @Nested
+    class Third {
+        @Test
+        void runs(WireMockRuntimeInfo info) {
+            assertDoesNotThrow(info::getHttpBaseUrl);
+        }
+    }
+}

